cli_bot.py

```diff
@@ -0,0 +1,433 @@
+            args: Command doesn't expect any args, list should be empty.
+
+        Returns:
+            Command output.
+        """
+        command_output = ""
+        if args:
+            command_output += (
+                "Warning: Command doesn't expect any arguments. "
+                f"Received: {' '.join(args)}\n"
+            )
+
+        command_output += "All Records: "
+        for record in self._address_book.values():
+            command_output += f"\n{record}"
+
+        return command_output
+
+    @input_error(error_msg_base="Command execution failed")
+    def execute_command(self, command: str, args: list[str]) -> str:
+        if command not in self.supported_commands:
+            raise CommandNotSupported(f"command '{command}' is not supported!")
+
+        command_handler = self.supported_commands[command]
+        return command_handler(*args)
+
+    @input_error(error_msg_base="Command 'add-address' failed")
+    def add_address(self, *args: str):
+        """Add address to already existing record.
+
+        Args:
+            args: List with username and address.
+
+        Returns:
+            Command output
+
+        Raises:
+            CommandOperationalError if user does not exist.
+            ValueError if date is invalid.
+        """
+        if len(args) != 2:
+            raise CommandOperationalError(
+                "command expects an input of two arguments: username and address, separated by a space. "
+                f"Received: {' '.join(args)}"
+            )
+        username, address_str = args
+        try:
+            record = self._address_book.find(username)
+
+        except KeyError as e:
+            raise CommandOperationalError(
+                f"user with username {username} doesn't exist. "
+                f"If you want to add address, please use 'add-address' command."
+            ) from e
+
+        record.add_address(address_str)
+        return f"Contact {username} updated with address: {address_str}."
+    
+    @input_error(error_msg_base="Command 'add-email' failed")
+    def add_email(self, *args: str):
+        """Add email to already existing record.
+
+        Args:
+            args: List with username and email.
+
+        Returns:
+            Command output
+
+        Raises:
+            CommandOperationalError if user does not exist.
+            ValueError if date is invalid.
+        """
+        if len(args) != 2:
+            raise CommandOperationalError(
+                "command expects an input of two arguments: username and email, separated by a space. "
+                f"Received: {' '.join(args)}"
+            )
+        username, email_str = args
+        try:
+            record = self._address_book.find(username)
+
+        except KeyError as e:
+            raise CommandOperationalError(
+                f"user with username {username} doesn't exist. "
+                f"If you want to add email, please use 'add-email' command."
```

Comment:
This message looks sus, he already calls this command, maybe this supposed to be reference to 'add' command to add a user that's being looked here ?